### DISCUSSION TITLE
Detached switch not recognised  by rule

### DISCUSSION THREAD

**Discussion by jarmilitao** [arendst/Tasmota#22646] [Support request]
Hi all,

On my tasmota I have a detached switch (SwitchMode4 15) and inserted the rule bellow:

Rule1 on Mqtt#State piscina/cmnd/switch4 do Backlog Status 1; MQTT publish piscina/tele/RESULT {"Switch4":"%Switch4%"} End

But when it receives a 'piscina/cmnd/switch4' message it apears on the console the following:

11:47:14.432 CMD: Grp 0, Cmd 'SWITCH', Idx 4, Len 0, Pld -99, Data ''
11:47:14.438 MQT: piscina/stat/RESULT = {"Command":"Unknown","Input":"SWITCH4"}

What am I doing wrong? Can someone, please, help me?

T.I.A.
J.A.R.Militão

**Comment by barbudor** [comment]
There is no such command 'switch4'
What are you trying to achieve ?

**Reply by barbudor** [reply]
And the syntax of your rule is pure hallucination
Have you used chatgpt instead of reading the docs ?

**Reply by jarmilitao** [reply]
I admit that sevelar atempts to find a way to  configure Tasmota as I need I even used  Gemini.

What I am trying to achieve is to have a switch (Tasmota switch1) that controls the power relay. But I also have another switch (Tasmota switch4) wich will sense thet power satus.


![simplified](https://github.com/user-attachments/assets/7269dfbd-a37b-49d1-8c71-b7d62c23bd46)

I managed to config the Tasmota switch4 as a detached switch but when homeassistant restarts the status of this switch is not reported with the same format as the one for switch1.

Is this understandable?
![Captura 1](https://github.com/user-attachments/assets/c69d396c-e041-42e5-ae43-2408f4fa2a40)

Thanks.

**Reply by sfromis** [reply]
Since your title is about rules not working for a "detached" Switch - the docs explains that SwitchMode 15 is:
> Send **only** MQTT message on switch change. This will stop the switch from controlling power outputs, and **you get no state values for rules**.

My emphasis, to explain why rules cannot work for a Switch input configured that way. If you want to catch all Switch state values with rules, just use one of the normal SwitchMode settings instead, as the presence of rules to trigger on the state changes will inherently decouple the default actions.

Of course, it is not clear what you want to process with rules and what you want to process with HA, which makes a difference to what config works for you.

**Reply by barbudor** [reply]
Switchmode15 is meant to decouple the switch from a relay of the same index and instead publish switch activity as MQTT messages
So if you just want to send MQTT meassage on switch activity, it is perfect and doesn't need rules.

Besides your template is not recommended as you should not jump indexes on compoments.
Each component (Relay, button, switch) should start at 1 and increment by 1 without "jumping"
As naturaly Tasmota will "couple" a switch of button with a relay of the same index, this is where "decoupling" is used to change the default behavior
As the doc states you can decouple a switch
- with switchmode15 to post switch activity to MQTT
- with rules to capture switch activity for doing somehting else

**Reply by sfromis** [reply]
`SetOption114` also exist to decouple switches, but works differently from SwitchMode 15. Strictly speaking, the coupling is not by same index, but solely based on the ordering, which especially becomes important in more combined setups.
